### DISCUSSION TITLE
Shared configuration object vs array confusing

### DISCUSSION THREAD

**Discussion by mauron85** [eslint/eslint#18583] [Ideas]
Sorry for being maybe overly pedantic. But I find this config duality object vs array extra confusing.

For example when I browse exported config eslint-all or eslint-recommended (https://www.npmjs.com/package/@eslint/js?activeTab=code). Both are exported as object.

On other hand plugin authors seems to have free choice, if they export their configs as object or array.

This duality even described in the docs:

>Shareable configs are simply npm packages that export a configuration object or array. To start, [create a Node.js module](https://docs.npmjs.com/getting-started/creating-node-modules) like you normally would.

Then consumer of such plugin needs to either spread array, or put add just imported object into configuration array.

Why not just make all configs even the defaults one array and only support arrays as shared / exported configuration?

**Comment by nzakas** [comment]
> Why not just make all configs even the defaults one array and only support arrays as shared / exported configuration?

Because we don't have any control over how you choose to load shareable configs. In the old config system, we were basically taking over the role of `require()` and could enforce exact what was exported from packages. In the new config system, you're on your own to `require()` or `import()` or even define other configs right in the same file. As a result, there is no way to force a convention. Even if we were to recommend something, most people won't read it, so it won't do much good. People might export a function that they expect you to call -- we just don't have any control over that.
